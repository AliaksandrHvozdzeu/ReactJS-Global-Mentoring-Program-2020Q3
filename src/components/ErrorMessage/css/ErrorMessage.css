.input-error {
    font-size: 0.8rem;
    font-weight: 100;
    color: var(--white);
    text-transform: none;
    margin-top: -20px;
    margin-bottom: 20px;
}
