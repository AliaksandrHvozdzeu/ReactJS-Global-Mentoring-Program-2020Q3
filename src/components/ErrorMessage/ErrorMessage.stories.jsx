import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import ErrorMessage from './js/ErrorMessage';
import './css/ErrorMessage.css';
import '../../App.css';

export default {
  title: 'ErrorMessage',
  decorators: [withKnobs],
};

export const Default = () => (
  <ErrorMessage error="Error message" />
);
