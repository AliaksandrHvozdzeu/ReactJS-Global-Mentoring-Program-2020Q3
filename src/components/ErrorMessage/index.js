import ErrorMessage from './js/ErrorMessage';

export default ErrorMessage;
