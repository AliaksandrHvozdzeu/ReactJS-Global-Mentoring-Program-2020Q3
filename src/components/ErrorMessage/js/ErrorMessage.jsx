import React from 'react';
import PropTypes from 'prop-types';
import '../css/ErrorMessage.css';

export default function ErrorMessage({ error }) {

  return (
    <>
      {error !== '' && <p className="input-error">{error}</p>}
    </>
  );

}

ErrorMessage.propTypes = {
  error: PropTypes.string,
};

ErrorMessage.defaultProps = {
  error: '',
};
