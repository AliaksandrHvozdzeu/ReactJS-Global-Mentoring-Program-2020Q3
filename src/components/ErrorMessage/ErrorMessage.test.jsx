import React from 'react';
import rerender from 'react-test-renderer';
import ErrorMessage from './js/ErrorMessage';

describe('when ErrorMessage', () => {
  test('then snapshot created', () => {
    const error = 'Title is required.';
    const component = rerender.create(<ErrorMessage error={error}/>);
    const tree = component.toJSON();
    expect(component.root.findByType('p').text()).toEqual("Title is required.");
    expect(tree).toMatchSnapshot();
  });

});
