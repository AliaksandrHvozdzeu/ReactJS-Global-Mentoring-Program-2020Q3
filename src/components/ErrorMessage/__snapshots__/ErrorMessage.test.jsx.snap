// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when ErrorMessage then snapshot created 1`] = `
<p
  className="input-error"
>
  Title is required.
</p>
`;
